file_repository/file_connexion.py

```diff
@@ -40,19 +42,17 @@ def open_and_close_connection(func):
     """
     @functools.wraps(func)
     def wrapper(self, *args, **kwargs):
-        if self.persistant:
-            if not self.connection:
-                self.connect()
-            return func(self, *args, **kwargs)
-        else:
+        if self.persistant and not self.connection:
```

Comment:
ummm, do you want me to correct this typo? It was there already and this it is a public attribute, I'm not sure it is safe to rename it.